analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,282 @@
+        val workingDir = definitionFile.parentFile
+        val (user, project, ref) = getProjectInfoFromVcs(workingDir)
+        return listOf(
+            ProjectAnalyzerResult(
+                project = Project(
+                    id = Identifier(
+                        type = managerName,
+                        namespace = user.orEmpty(),
+                        name = project.orEmpty(),
+                        version = ref.orEmpty()
+                    ),
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    declaredLicenses = sortedSetOf(),
+                    vcs = VcsInfo.EMPTY,
+                    vcsProcessed = processProjectVcs(workingDir, VcsInfo.EMPTY),
+                    scopes = sortedSetOf(),
+                    homepageUrl = ""
+                ),
+                packages = parseCarthageDependencies(definitionFile)
+            )
+        )
+    }
+
+    /**
+     * As the Carthage.resolved file does not provide any project information, trying to retrieve some from VCS.
+     */
+    private fun getProjectInfoFromVcs(workingDir: File): Triple<String?, String?, String?> {
+        val workingTree = VersionControlSystem.forDirectory(workingDir)
+        val vcsInfo = workingTree?.getInfo() ?: VcsInfo.EMPTY
+        val normalizedVcsUrl = normalizeVcsUrl(vcsInfo.url)
+        val vcsHost = VcsHost.fromUri(URI(normalizedVcsUrl))
+        return Triple(
+            vcsHost?.getUserOrOrganization(normalizedVcsUrl),
+            vcsHost?.getProject(normalizedVcsUrl),
+            vcsInfo.resolvedRevision
+        )
+    }
+
+    private fun parseCarthageDependencies(definitionFile: File): SortedSet<Package> {
+        val dependencyLines = definitionFile.readLines()
+        val workingDir = definitionFile.parent
+        val packages = sortedSetOf<Package>()
+        dependencyLines.forEach { line ->
+            if (line.isNotEmpty() && !isComment(line)) {
+                val (carthageType, id, ref) = parseCarthageDependencyLine(line)
+                when (carthageType) {
+                    CarthageDependencyType.GITHUB -> {
+                        // ID consists of github username/project or a github enterprise URL.
+                        if (isGithubUserAndProject(id)) {
+                            packages.add(createPackageFromGithubUserAndProject(id, ref))
+                        } else {
+                            packages.add(createPackageFromGenericGitUrl(id, ref))
+                        }
+                    }
+                    CarthageDependencyType.GIT -> {
+                        // ID is a generic git URL (https, git or ssh) or a file path to a local repository.
+                        if (isFilePath(workingDir, id)) {
+                            // TODO Carthage supports local git repositories.
+                            //  The downloader should be extended to support `file://` downloads.
```

Comment:
I wonder if 'file:///' URLs should be included into analyzer results at all, because that would introduce the assumption that the analyzer.result can be further examined (downloaded / scanned) only on the same machine where it got analyzed.

Reply:
That's a good point, but my argument for keeping `file://` URLs was that they are not really different from `http://` URLs where content has been removed from the server side, and thus they're broken. And it's probably good to let the scanner / downloader run into these broken links to highlight where curations are necessary.